src/contracts/GPv2Settlement.sol

```diff
@@ -231,4 +247,18 @@ contract GPv2Settlement {
 
         inTransfer.amount = inTransfer.amount.add(order.tip);
     }
+
+    /// @dev Compute a unique identifier that represents a user order.
+    /// @param orderDigest The unique digest associated to the parameters of an
+    /// order (an instance of the Order struct in the [`GPv2Encoding`] library).
+    /// The order digest is the (unpacked) hash of all entries in the order in
+    /// which they appear.
+    /// @param owner The address of the user to assign to the order.
+    function orderId(bytes32 orderDigest, address owner)
```

Comment:
This will probably need to also include the validity for the order gas refund to work. But I don't think we should add it here, and can add that when order refunds are implemented.

Reply:
naming-nit: Maybe we should call this a UID or UUID to indicate it is truly unique?
```suggestion
    function orderUid(bytes32 orderDigest, address owner)
```